db/data/20210303102617_remove_royal_accademy_of_engineering_and_british_accademy_legacy_data.rb

```diff
@@ -0,0 +1,29 @@
+# Run me with `rails runner db/data/20210303102617_remove_royal_accademy_of_engineering_and_british_accademy_legacy_data.rb`
+#
+# Put your Ruby code here
+def remove_all_data_set_for_organisation_id(organisation_id)
+  # collect all activities
+  activities = Activity.where(organisation_id: organisation_id).or(Activity.where(extending_organisation_id: organisation_id))
+  activity_ids = activities.pluck(:id)
+
+  # destroy all transactions
+  transactions = Transaction.where(parent_activity_id: activity_ids)
+  transactions.destroy_all
+
+  # destroy all forecasts
+  forecasts = PlannedDisbursement.where(parent_activity_id: activity_ids)
+  forecasts.destroy_all
+
+  # destroy all budgets
+  budgets = Budget.where(parent_activity_id: activity_ids)
+  budgets.destroy_all
+
+  # destroy all activites
+  activities.destroy_all
+end
+
+royal_accademy_of_engineering_id = Organisation.find_by(name: "Royal Academy of Engineering", iati_reference: "GB-CHC-293074")&.id
+british_accademy_id = Organisation.find_by(name: "British Academy", iati_reference: "GB-COH-RC000053")&.id
```

Comment:
Should these raise an error if the given records are not found? As written, this script could silently do nothing if the expected organisation records are not loaded.

Reply:
It's a fair point, I originally wrote it with the IDs of the organisation in the data migraton as this is explicitally to be run in production only, where the IDs are known. 

Then I changed it as testing with known IDs became too complicated, I should have changed it back at the point we decided that a migration like this one, that removes a bunch of data, probably doesn't need a test to give us confidence, or rather writting test for it feels like too much effort for the confidence it would bring, we felt running it against a copy of the production database would deliver more confidence.

If I switch back to the known IDs I can guarantee the code will execute, how does that sit with you?

Reply:
I think using the names rather than IDs makes the code clearer, but we could make the script error out if the required records are not found?